tests/data/test_unsupervised_sampler.py

```diff
@@ -19,65 +19,113 @@
-        batch_size = 4
-        sampler = UnsupervisedSampler(G=line_graph, length=2, number_of_walks=2)
-        batches = sampler.run(batch_size)
 
-        # check batch sizes
-        assert len(batches) == np.ceil(len(line_graph.nodes()) * 4 / batch_size)
-        for ids, labels in batches[:-1]:
-            assert len(ids) == len(labels) == batch_size
+def test_run_batch_sizes(line_graph):
+    batch_size = 4
+    sampler = UnsupervisedSampler(G=line_graph, length=2, number_of_walks=2)
+    batches = sampler.run(batch_size)
 
-        # last batch can be smaller
-        ids, labels = batches[-1]
-        assert len(ids) == len(labels)
-        assert len(ids) <= batch_size
+    # check batch sizes
+    assert len(batches) == np.ceil(len(line_graph.nodes()) * 4 / batch_size)
+    for ids, labels in batches[:-1]:
+        assert len(ids) == len(labels) == batch_size
 
-    def test_run_context_pairs(self, line_graph):
-        batch_size = 4
-        sampler = UnsupervisedSampler(G=line_graph, length=2, number_of_walks=2)
-        batches = sampler.run(batch_size)
+    # last batch can be smaller
+    ids, labels = batches[-1]
+    assert len(ids) == len(labels)
+    assert len(ids) <= batch_size
 
-        grouped_by_target = defaultdict(list)
 
-        for ids, labels in batches:
-            for (target, context), label in zip(ids, labels):
-                grouped_by_target[target].append((context, label))
+def test_run_context_pairs(line_graph):
+    batch_size = 4
+    sampler = UnsupervisedSampler(G=line_graph, length=2, number_of_walks=2)
+    batches = sampler.run(batch_size)
 
-        assert len(grouped_by_target) == len(line_graph.nodes())
+    grouped_by_target = defaultdict(list)
 
-        for target, sampled in grouped_by_target.items():
-            # exactly 2 positive and 2 negative context pairs for each target node
-            assert len(sampled) == 4
+    for ids, labels in batches:
+        for (target, context), label in zip(ids, labels):
+            grouped_by_target[target].append((context, label))
 
-            # since each walk has length = 2, there must be an edge between each positive context pair
-            for context, label in sampled:
-                if label == 1:
-                    assert context in set(line_graph.neighbors(target))
+    assert len(grouped_by_target) == len(line_graph.nodes())
+
+    for target, sampled in grouped_by_target.items():
+        # exactly 2 positive and 2 negative context pairs for each target node
+        assert len(sampled) == 4
+
+        # since each walk has length = 2, there must be an edge between each positive context pair
+        for context, label in sampled:
+            if label == 1:
+                assert context in set(line_graph.neighbors(target))
+
+
+def test_walker_uniform_random(line_graph):
+    length = 3
+    number_of_walks = 2
+    batch_size = 4
+
+    walker = UniformRandomWalk(line_graph, n=number_of_walks, length=length)
```

Comment:
What's this testing? It looks like it's almost testing that `UnsupervisedSampler(..., walker=None)` uses `UniformRandomWalk` internally, but it's not quite doing the validation for that. It also has a lot of overlap with `CustomWalker`?

Reply:
It's just testing that the batches are consistent with the walker object provided instead of the default values for `number_of_walks` and `length` for `UnsupervisedSampler`. I guess the custom walker test also does a similar job, but I thought I'd include one that uses one of our actual `RandomWalk` classes.

Reply:
Ah, I see. Makes sense. Maybe a brief comment to provide context?